File: src/test/java/com/flexpoker/game/command/aggregate/tablebalancer/TableBalancerExceptionTest.java
package com.flexpoker.game.command.aggregate.tablebalancer;

import static com.flexpoker.game.command.aggregate.tablebalancer.TableBalancerTestUtils.createDefaultChipMapForSubjectTable;
import static com.flexpoker.game.command.aggregate.tablebalancer.TableBalancerTestUtils.createTableToPlayersMap;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.Collections;
import java.util.UUID;

import org.junit.jupiter.api.Test;

import com.flexpoker.exception.FlexPokerException;
import com.flexpoker.game.command.aggregate.TableBalancer;

public class TableBalancerExceptionTest {

    @Test
    void testSingleTableOnePlayer() {
        var subjectTableId = UUID.randomUUID();
        var tableToPlayersMap = createTableToPlayersMap(subjectTableId, 1);

        var tableBalancer = new TableBalancer(UUID.randomUUID(), 2);
        assertThrows(FlexPokerException.class,
                () -> tableBalancer.createSingleBalancingEvent(subjectTableId, Collections.emptySet(),
                        tableToPlayersMap, createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap)));
    }

    @Test
    void testTwoTablesOnePlayer() {
        var subjectTableId = UUID.randomUUID();
        var tableToPlayersMap = createTableToPlayersMap(subjectTableId, 1, 0);

        var tableBalancer = new TableBalancer(UUID.randomUUID(), 2);
        assertThrows(FlexPokerException.class,
                () -> tableBalancer.createSingleBalancingEvent(subjectTableId, Collections.emptySet(),
                        tableToPlayersMap, createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap)));
    }

    @Test
    void testAllTablesEmpty() {
        var subjectTableId = UUID.randomUUID();
        var tableToPlayersMap = createTableToPlayersMap(subjectTableId, 0, 0);

        var tableBalancer = new TableBalancer(UUID.randomUUID(), 2);
        assertThrows(FlexPokerException.class,
                () -> tableBalancer.createSingleBalancingEvent(subjectTableId, Collections.emptySet(),
                        tableToPlayersMap, createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap)));
    }

}

File: src/test/kotlin/com/flexpoker/game/command/aggregate/tablebalancer/TableBalancerExceptionTest.kt
package com.flexpoker.game.command.aggregate.tablebalancer

import com.flexpoker.exception.FlexPokerException
import com.flexpoker.game.command.aggregate.createSingleBalancingEvent
import com.flexpoker.test.util.UnitTestClass
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.util.UUID

@UnitTestClass
class TableBalancerExceptionTest {

    @Test
    fun testSingleTableOnePlayer() {
        val subjectTableId = UUID.randomUUID()
        val tableToPlayersMap = TableBalancerTestUtils.createTableToPlayersMap(subjectTableId, 1)
        assertThrows(FlexPokerException::class.java) {
            createSingleBalancingEvent(UUID.randomUUID(), 2, subjectTableId, emptySet(), tableToPlayersMap,
                TableBalancerTestUtils.createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap))
        }
    }

    @Test
    fun testTwoTablesOnePlayer() {
        val subjectTableId = UUID.randomUUID()
        val tableToPlayersMap = TableBalancerTestUtils.createTableToPlayersMap(subjectTableId, 1, 0)
        assertThrows(FlexPokerException::class.java) {
            createSingleBalancingEvent(UUID.randomUUID(), 2, subjectTableId, emptySet(), tableToPlayersMap,
                TableBalancerTestUtils.createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap))
        }
    }

    @Test
    fun testAllTablesEmpty() {
        val subjectTableId = UUID.randomUUID()
        val tableToPlayersMap = TableBalancerTestUtils.createTableToPlayersMap(subjectTableId, 0, 0)
        assertThrows(FlexPokerException::class.java) {
            createSingleBalancingEvent(UUID.randomUUID(), 2, subjectTableId, emptySet(), tableToPlayersMap,
                TableBalancerTestUtils.createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap))
        }
    }

}